tests/tests-jobqueue.cpp

```diff
@@ -0,0 +1,16 @@
+#include <gtest/gtest.h>
+#include <thread>
+#include <job-queue.h>
+#include "common.h"
+
+TEST(jobqueue, complete_job) {
+    std::thread job_thread = std::thread(&JobQueue::Process, &JobQueue::GetInstance());
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    std::atomic<bool> job_completed = false;
+    ASSERT_TRUE(JobQueue::GetInstance().AddJob([&](){
+        job_completed = true;
+    }));
+    JobQueue::GetInstance().Stop();
+    job_thread.join();
+    ASSERT_TRUE(job_completed.load());
```

Comment:
I wonder if this is sufficient. I feel like we should design a unit test that runs several jobs then waits for a while, then runs more jobs. Then checks to make sure they are all completed.
Hopefully this would tell us if the job queue has the potential to segfault.. Though it would not test against spurious wakeups